packages/controller/main.js

```diff
@@ -1191,22 +1192,32 @@ function setIPs(ipList) {
     }
 }
 
-function extendObjects(tasks, callback) {
-    if (!tasks || !tasks.length) {
-        return typeof callback === 'function' && callback();
-    }
-    const task = tasks.shift();
-    const state = task.state;
-    if (state !== undefined) {
-        delete task.state;
-    }
-    objects.extendObject(task._id, task, () => {
+/**
+ * Extends objects, optionally you can provide a state at each task (does not throw)
+ * @param {Record<string, any>[]} tasks
+ * @returns {Promise<void>}
+ */
+async function extendObjects(tasks) {
+    for (const task of tasks) {
+        const state = task.state;
+        if (state !== undefined) {
+            delete task.state;
+        }
+
+        try {
+            await objects.extendObjectAsync(task._id, task);
+        } catch {
+            // ignore
```

Comment:
I think we should not setState on error (yes this was ignored before too, but better so or?!)

Reply:
Good point we don’t want States w/o objects, so yes